packages/governance/src/binaryBallotCounter.js

```diff
@@ -23,60 +30,71 @@ const makeQuorumCounter = quorumThreshold => {
   return Far('checker', { check });
 };
 
-// Exported for testing purposes
-const makeBinaryBallotCounter = (
-  question,
-  positions,
-  threshold,
-  tieOutcome = undefined,
-  closingRule,
-  instance,
-) => {
+const validateBinaryBallotSpec = ballotSpec => {
+  looksLikeBallotSpec(ballotSpec);
+
+  const positions = ballotSpec.positions;
   assert(
     positions.length === 2,
     X`Binary ballots must have exactly two positions. had ${positions.length}: ${positions}`,
   );
-  assert.typeof(question, 'string');
-  assert.typeof(positions[0], 'string');
-  assert.typeof(positions[1], 'string');
-  if (tieOutcome) {
-    assert(
-      positions.includes(tieOutcome),
-      X`The default outcome on a tie must be one of the positions, not ${tieOutcome}`,
-    );
-  }
-
-  const template = buildBallot(
-    ChoiceMethod.CHOOSE_N,
-    question,
-    positions,
-    1,
-    instance,
+  assert(
+    ballotSpec.maxChoices <= positions.length,
+    X`Choices must not exceed length`,
   );
-  const ballotDetails = template.getDetails();
 
+  looksLikeQuestionForType(ballotSpec.electionType, ballotSpec.question);
+
+  assert(
+    ballotSpec.maxChoices === 1,
+    X`Can only choose 1 item on a binary ballot`,
+  );
   assert(
-    ballotDetails.method === ChoiceMethod.CHOOSE_N,
-    X`Binary ballot counter only works with CHOOSE_N`,
+    ballotSpec.method === ChoiceMethod.CHOOSE_N,
+    X`${ballotSpec.method} must be CHOOSE_N`,
   );
+  // We don't check the quorumRule or quorumThreshold here. The quorumThreshold
+  // is provided by the Registrar that creates this ballotCounter, since only it
+  // can translate the quorumRule to a required number of votes.
+};
+
+// Notice that BinaryBallotCounter is designed to run as a Zoe contract. The
+// business part of the contract is extracted here so it can be tested
+// independently. The standard Zoe start function is at the bottom of this file.
+
+/** @type {BuildBallotCounter} */
+const makeBinaryBallotCounter = (ballotSpec, threshold, instance) => {
+  validateBinaryBallotSpec(ballotSpec);
+
+  const ballot = buildBallot(ballotSpec, instance);
+  const details = ballot.getDetails();
+  // @ts-ignore typescript is confused here. Dunno why. The declarations clearly
+  // say Ballot.getDetails() returns a ballotDetails, which has a handle.
+  const { handle } = details;
```

Comment:
It looks like I may be wrong about how to do subtyping in JS. When I hover over `details` TS (via vscode) tells me it is a `BallotSpec` rather than a `BallotDetails`.

attn @michaelfig @mhofman how do I declare that one type is like another, but with the following additional properties?

Reply:
> attn @michaelfig @mhofman how do I declare that one type is like another, but with the following additional properties?

```js
/** @typedef {Object} BaseProperties
  * @property {Type} name1
  *...
*/
/** @typedef {Object} AdditionalProperties
  * @property {Type} name2
  *...
*/
// Ask for the intersection (&) of the two types.  This doesn't strictly need
// typedefs to work, it can use inline types as well.
/** @typedef {BaseProperties & AdditionalProperties} ExtendedType */
```

Reply:
@michaelfig, I thought I learned something from @erights' original suggestion that sub-typing could be done by replacing the `Object` in `/** @typedef {Object} AdditionalProperties`.  In order to help me unlearn the lesson, can you say whether there's ever a reason to use something other than `{Object}` in the type position of a `@typedef`?

Reply:
> can you say whether there's ever a reason to use something other than {Object} in the type position of a @typedef

The reason to have `Object` is so that subsequent `@property {Type} name` lines affect the `@typedef`.  If you want fully inline the types instead of describing with `@property`, then that's the time you'd use something instead of `Object` like:

```js
/**
 * @typedef {{ name: Type }} Named
 */
```